qt/component-box.cpp

```diff
@@ -125,3 +125,18 @@ void ComponentBox::sortByScore()
 {
     as_component_box_sort_by_score(d->m_cbox);
 }
+
+ComponentBox::iterator ComponentBox::erase(iterator it)
+{
+    GPtrArray *array = as_component_box_array(it.data->d->m_cbox);
+    g_ptr_array_remove_index(array, it.index);
+    return it;
+}
+
+void ComponentBox::operator+=(const ComponentBox& other)
+{
+    for (uint i = 0; i < as_component_box_len(other.d->m_cbox); i++) {
+        AsComponent *c = as_component_box_index(other.d->m_cbox, i);
+        as_component_box_add(d->m_cbox, c, nullptr);
```

Comment:
Is there any way that we could propagate an error here? Currently, components may silently not be added... (I don't see a sane way to do that, so this might just be as good as it gets...)

Reply:
I'll add a warning, then we can decide to extend the API if we see it's a real problem.